compiler/IREmitter/SymbolBasedIntrinsics.cc

```diff
@@ -756,18 +756,99 @@ class TPrivateCompiler_compilerVersion : public SymbolBasedIntrinsicMethod {
     };
 } TPrivateCompiler_compilerVersion;
 
+class Thread_squareBrackets : public CallCMethod {
```

Comment:
This is the first time we're subclassing `CallCMethod`. I guess that's probably fine?

(I'm generally not a fan of subclassing for the purpose of code reuse, in favor of helper functions, but this seems ~fine I guess.)